### PR TITLE
Fuse Fixes

### PR DESCRIPTION
**Pull request by sidsenkumar11**
- Dirname fix to output "/" on paths like "/test"
- Read returns 0 if offset is past file
- Added fuse_access() function
- fuse_chown() doesn't set uid/gid if the values are max values of a 2 or 4 byte int.
- Changed INode's mode, uid, and gid to be 32-bit values.
- fuse_getattr() binary OR's a file's mode with its file type.
- Default to using FileStorage in mkfs.cpp and fuse.cpp. fuse.cpp also expects the disk to have 788496 blocks and be located at /dev/vdc.

### CODE REVIEW

File: src/lib/Filesystem.cpp [hunk 1]
```diff
@@ -593,6 +594,9 @@ std::string Filesystem::dirname(const char* path_cstring) {
     dir_path = "/" + folder_names.top() + dir_path;
     folder_names.pop();
   }
+	if (dir_path.length() == 0) {
```
Comment (xavierholt) [inline src/lib/Filesystem.cpp]: Can we change these to spaces?

File: src/fuse.cpp [hunk 2]
```diff
@@ -431,10 +452,11 @@ int main(int argc, char** argv) {
 
   // Default ~32GB disk
   // TODO: Read value from argv
-  uint64_t nblocks = 1 + 10 + (1 + 512) + (1 + 512 + 512*512) + (1 + 2 + 512*2 + 512*512*2);
+  // uint64_t nblocks = 1 + 10 + (1 + 512) + (1 + 512 + 512*512) + (1 + 2 + 512*2 + 512*512*2);
+  uint64_t nblocks =  788496;
 
   // Instantiate objects for filesystem
-  Storage *disk = new MemoryStorage(nblocks);
+  Storage *disk = new FileStorage("/dev/vdc", nblocks);
```
Comment (xavierholt) [inline src/fuse.cpp]: Hardcoding is gross, but oh well.  Maybe we'll get a CLI eventually...

Comment (sidsenkumar11) [reply src/fuse.cpp]: Yeah.... I was just getting annoyed with inserting this every branch.

File: src/lib/Filesystem.cpp (hunk 1 above)
Comment (sidsenkumar11) [reply src/lib/Filesystem.cpp]: Can do.

**Pull request merged**